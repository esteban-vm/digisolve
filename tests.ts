import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
