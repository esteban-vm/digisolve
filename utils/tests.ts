import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
