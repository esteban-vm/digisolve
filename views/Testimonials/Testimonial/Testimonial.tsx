import type { ImageComponent } from '@/types'
import Image from 'next/image'
import { styled } from '@/styles'

const TestimonialComponent: ImageComponent = ({ src, alt, ...rest }) => {
  return (
    <article {...rest}>
      <blockquote className='testimonial__quote'>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. At, praesentium.
      </blockquote>
      <cite className='testimonial__box'>
        <Image src={src} alt={alt} className='testimonial__photo' />
        {alt}
      </cite>
    </article>
  )
}

const Testimonial = styled(TestimonialComponent)`
  text-align: left;
  margin-bottom: 50px;
  user-select: none;

  .testimonial {
    &__quote {
      padding: 2%;
      font-style: italic;
      line-height: 145%;
      position: relative;
      margin-top: 40px;

      ::before {
        content: '\\201C';
        display: block;
        font-size: 500%;
        position: absolute;
        left: -5px;
        top: -5px;
      }

      ::first-letter {
        text-transform: capitalize;
      }
    }

    &__box {
      display: block;
      margin-top: 25px;
      font-size: 90%;
      text-transform: capitalize;
    }

    &__photo {
      border-radius: 50%;
      height: 20%;
      vertical-align: middle;
      margin-right: 10px;
    }
  }
`

export default Testimonial
