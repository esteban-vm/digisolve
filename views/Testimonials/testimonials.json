[
  {
    "image": "/img/client-1.jpg",
    "text": "Aaron Cruz",
    "height": 72,
    "width": 72
  },
  {
    "image": "/img/client-2.jpg",
    "text": "Jeff Jones",
    "height": 72,
    "width": 72
  },
  {
    "image": "/img/client-3.jpg",
    "text": "Helen Smith",
    "height": 72,
    "width": 72
  }
]
