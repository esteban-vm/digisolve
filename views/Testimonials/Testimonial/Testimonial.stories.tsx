import { type Meta, type StoryObj, within, userEvent } from '@/stories'
import { Grid } from '@/components'
import { styled } from '@/styles'
import { testimonials } from '../Testimonials'
import Testimonial from './Testimonial'

const Wrapper = styled.div`
  text-align: center;
  color: var(--color-white);
  width: 75vw;

  background-image: linear-gradient(var(--color-overlay3), var(--color-overlay4)), url('/img/testimonial-bg.jpg');
  background-size: cover;
  background-attachment: fixed;

  article {
    padding: 1rem 0;
  }
`

export default {
  argTypes: {
    src: {
      control: false,
    },
  },
  component: Testimonial,
  title: 'Views/Testimonials/Testimonials',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Grid.Row>
            <Grid.Col isOneThird>&nbsp;</Grid.Col>
            <Grid.Col isOneThird>
              <Story {...context} />
            </Grid.Col>
            <Grid.Col isOneThird>&nbsp;</Grid.Col>
          </Grid.Row>
        </Wrapper>
      )
    },
  ],
  parameters: {
    layout: 'centered',
  },
  play({ canvasElement }) {
    const canvas = within(canvasElement)
    const photo = canvas.getByRole('img')
    userEvent.hover(photo)
  },
} as Meta<typeof Testimonial>

const Examples: StoryObj<typeof Testimonial>[] = []
for (const testimonial of testimonials) Examples.push({ args: testimonial })

export const Example1 = Examples[0]
export const Example2 = Examples[1]
export const Example3 = Examples[2]
