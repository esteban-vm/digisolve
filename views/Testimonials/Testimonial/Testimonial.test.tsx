import { render, cleanup, screen, create } from '@/tests'
import testimonials from '../testimonials.json'
import Testimonial from './Testimonial'

describe('ðŸ§ª TESTIMONIAL:', () => {
  const [testTestimonial] = testimonials

  describe('display tests:', () => {
    beforeEach(() => {
      render(<Testimonial {...testTestimonial} />)
    })

    afterEach(cleanup)

    it('should be accessible', () => {
      const article = screen.getByRole('article', { name: testTestimonial.text })
      expect(article).toBeInTheDocument()
    })

    describe('should be displayed:', () => {
      it('the photo', () => {
        const photo = screen.getByRole('img', { name: testTestimonial.text })
        expect(photo).toBeInTheDocument()
        expect(photo).toBeVisible()
      })

      it('the name', () => {
        const name = screen.getByText(testTestimonial.text)
        expect(name).toBeInTheDocument()
        expect(name).toBeVisible()
      })

      it('the quote', () => {
        const quote = screen.getByText(/^lorem ipsum/i)
        expect(quote).toBeInTheDocument()
        expect(quote).toBeVisible()
      })
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<Testimonial {...testTestimonial} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
