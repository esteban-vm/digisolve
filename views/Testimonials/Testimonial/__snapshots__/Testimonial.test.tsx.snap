// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª TESTIMONIAL: style tests: should render with correct styles 1`] = `
.emotion-0 {
  text-align: left;
  margin-bottom: 50px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 .testimonial__quote {
  padding: 2%;
  font-style: italic;
  line-height: 145%;
  position: relative;
  margin-top: 40px;
}

.emotion-0 .testimonial__quote::before {
  content: '\\201C';
  display: block;
  font-size: 500%;
  position: absolute;
  left: -5px;
  top: -5px;
}

.emotion-0 .testimonial__quote::first-letter {
  text-transform: capitalize;
}

.emotion-0 .testimonial__box {
  display: block;
  margin-top: 25px;
  font-size: 90%;
  text-transform: capitalize;
}

.emotion-0 .testimonial__photo {
  border-radius: 50%;
  height: 20%;
  vertical-align: middle;
  margin-right: 10px;
}

<article
  className="emotion-0"
>
  <blockquote
    className="testimonial__quote"
  >
    Lorem ipsum dolor, sit amet consectetur adipisicing elit. At, praesentium.
  </blockquote>
  <cite
    className="testimonial__box"
  >
    <img
      alt="Aaron Cruz"
      className="testimonial__photo"
      data-nimg="1"
      decoding="async"
      height={40}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={40}
    />
    Aaron Cruz
  </cite>
</article>
`;
